buildSrc/src/main/java/io/servicetalk/internal/build/ExecutableBuilder.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright © 2020 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.internal.build;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+
+public final class ExecutableBuilder {
+
+    private ExecutableBuilder() {
+        // no instances
+    }
+
+    public static void buildUnixExecutable(String uberJarName, File outputFile) throws IOException {
+        prepareOutputFile(outputFile);
+        try(FileOutputStream execOutputStream = new FileOutputStream(outputFile)) {
+            execOutputStream.write(("#!/bin/sh\n" +
+                    "pushd $(dirname \"$0\") > /dev/null\n" +
+                    "exec java -jar " + uberJarName + " \"$@\"\n" +
+                    "popd > /dev/null\n").getBytes(StandardCharsets.US_ASCII));
+        }
+        finalizeOutputFile(outputFile);
+    }
+
+    public static void buildWindowsExecutable(String uberJarName, File outputFile) throws IOException {
+        prepareOutputFile(outputFile);
+        try(FileOutputStream execOutputStream = new FileOutputStream(outputFile)) {
+            execOutputStream.write(("@ECHO OFF\n" +
+                    "pushd %~dp0\n" +
+                    "java -jar " + uberJarName + " %*\n" +
+                    "popd\n").getBytes(StandardCharsets.US_ASCII));
```

Comment:
Should we use `\r\n` for windows?

Reply:
I tested with the `\n` but yes let me change to `\r\n`.